elasticdl/python/master/task_dispatcher.py

```diff
@@ -318,7 +322,7 @@ def report(self, request, success):
             if evaluation_task_completed:
                 self._evaluation_service.complete_task()
 
-            if self._callbacks_list.model.stop_training:
+            if success and self._callbacks_list.model.stop_training:
```

Comment:
Why do we check success here?
If the task always fail, will it enter a infinite loop?

Reply:
Yes, we have added a TODO here.
https://github.com/sql-machine-learning/elasticdl/blob/a7be465c73e27872db0dae70c1ffa5296970cee6/elasticdl/python/master/task_dispatcher.py#L301-L305

We may need to implement the TODO now.

Reply:
I have implemented the TO in #1854